source/common/http/conn_manager_impl.cc

```diff
@@ -921,9 +928,12 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(RequestHeaderMapPtr&& he
 
   // TODO if there are no filters when starting a filter iteration, the connection manager
   // should return 404. The current returns no response if there is no router filter.
-  if (protocol == Protocol::Http11 && hasCachedRoute()) {
+  if (hasCachedRoute()) {
     if (upgrade_rejected) {
       // Do not allow upgrades if the route does not support it.
+      state_.saw_connection_close_ = true;
```

Comment:
Can you add a comment on why we are setting this in this case but not other early reply cases?